pkg/api/v1/defaults.go

```diff
@@ -195,6 +195,28 @@ func SetDefaults_PodSpec(obj *PodSpec) {
 		obj.SchedulerName = DefaultSchedulerName
 	}
 }
+func SetDefaults_Affinity(affinity *Affinity) {
+	if affinity != nil {
+		// If pod anti-affinity is not nil, UPDATE default value of MaxPodsPerTopologyKey to 1.
+		if affinity.PodAntiAffinity != nil {
+			antiAffinity := affinity.PodAntiAffinity
+			for i := range antiAffinity.RequiredDuringSchedulingIgnoredDuringExecution {
+				term := &antiAffinity.RequiredDuringSchedulingIgnoredDuringExecution[i]
+				if term.MinMatchingPods < 1 {
+					term.MinMatchingPods = 1
+				}
+			}
+		}
+		// If pod affinity is not nil, SET value of MaxPodsPerTopologyKey to 1.
```

Comment:
Done for the comments.

I left the comments for "perferred", as it's the places I'm going to set default for preferred; so replaced logic with a line of comment.